addons/website_sale_delivery/controllers/main.py

```diff
@@ -22,6 +22,13 @@ def payment(self, **post):
 
         return super(WebsiteSaleDelivery, self).payment(**post)
 
+    @http.route()
+    def payment_transaction(self, *args, **kwargs):
+        order = request.website.sale_get_order()
+        if not order.is_all_service and (order.recompute_delivery_price or not order.delivery_set):
```

Comment:
```suggestion
        if not order.is_all_service and not order.delivery_set:
```
Shouldn't this be enough?

Reply:
You are right !